concept/answer/ConceptMapImpl.java

```diff
@@ -23,208 +23,175 @@
-    }
-
-    public ConceptMapImpl(Map<String, Concept> map, Explainables explainables) {
-        this.map = Collections.unmodifiableMap(map);
-        this.explainables = explainables;
-    }
-
-    public static ConceptMap of(AnswerProto.ConceptMap res) {
-        Map<String, Concept> variableMap = new HashMap<>();
-        res.getMapMap().forEach((resVar, resConcept) -> variableMap.put(resVar, ConceptImpl.of(resConcept)));
-        return new ConceptMapImpl(Collections.unmodifiableMap(variableMap), of(res.getExplainables()));
-    }
-
-    private static Explainables of(AnswerProto.Explainables explainables) {
-        Map<String, Explainable> relations = new HashMap<>();
-        explainables.getRelationsMap().forEach((var, explainable) -> {
-            relations.put(var, ExplainableImpl.of(explainable));
-        });
-        Map<String, Explainable> attributes = new HashMap<>();
-        explainables.getAttributesMap().forEach((var, explainable) -> {
-            attributes.put(var, ExplainableImpl.of(explainable));
-        });
-        Map<Pair<String, String>, Explainable> ownerships = new HashMap<>();
-        explainables.getOwnershipsMap().forEach((var, ownedMap) -> {
-            ownedMap.getOwnedMap().forEach((owned, explainable) -> {
-                ownerships.put(new Pair<>(var, owned), ExplainableImpl.of(explainable));
-            });
-        });
-        return new ExplainablesImpl(relations, attributes, ownerships);
+import static com.vaticle.typedb.client.jni.typedb_client.concept_map_equals;
+import static com.vaticle.typedb.client.jni.typedb_client.concept_map_get;
+import static com.vaticle.typedb.client.jni.typedb_client.concept_map_get_explainables;
+import static com.vaticle.typedb.client.jni.typedb_client.concept_map_get_values;
+import static com.vaticle.typedb.client.jni.typedb_client.concept_map_get_variables;
+import static com.vaticle.typedb.client.jni.typedb_client.concept_map_to_string;
+import static com.vaticle.typedb.client.jni.typedb_client.explainable_get_conjunction;
+import static com.vaticle.typedb.client.jni.typedb_client.explainable_get_id;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_equals;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_get_attribute;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_get_attributes_keys;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_get_ownership;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_get_ownerships_keys;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_get_relation;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_get_relations_keys;
+import static com.vaticle.typedb.client.jni.typedb_client.explainables_to_string;
+
+public class ConceptMapImpl extends NativeObject<com.vaticle.typedb.client.jni.ConceptMap> implements ConceptMap {
+    private final int hash;
```

Comment:
let's replace this with a proper (lazily cached) hash code using a computed hash over the streamed entries' hashes

let's apply this logic to all the `Answer` types